src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (newNode.prev != null) {
+            newNode.prev.next = newNode;
+        }
+        if (head == null) {
+            head = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index out of range");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node currentNode = findNode(index);
+        if (index == 0) {
+            Node newNode = new Node(null, value, currentNode);
+            currentNode.prev = newNode;
+            head = newNode;
+        } else {
+            Node newNode = new Node(currentNode.prev, value, currentNode);
+            currentNode.prev.next = newNode;
+            currentNode.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node currentNode = findNode(index);
+        return (T) currentNode.element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node currentNode = findNode(index);
+        T oldValue = (T) currentNode.element;
+        currentNode.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node currentNode = findNode(index);
+        return unlinkNode(currentNode);
     }
 
     @Override
     public boolean remove(T object) {
+        for (int i = 0; i < size; i++) {
+            if (object == findNode(i).element || object != null
+                    && object.equals(findNode(i).element)) {
+                remove(i);
+                return true;
+            }
+        }
```

Comment:
```suggestion
        for (Node<T> node = head; head != null; node = node.next) {
            if (node.value == value || value != null && value.equals(node.value)) {
               unlink(node);
               return true;
            }
        }
        return false;
```
using findNode(i) like this is not a good idea, because you invoke this method and iterate almost over the whole list 2 times for each current iteration
there is no need for this, because you can iterate over the actual nodes, and pass the node with the value that is to be removed to a unlink method